test/null_safe/change_notifier_provider_test.dart

```diff
@@ -356,4 +356,61 @@ void main() {
 
     expect(myNotifier.notifyListeners, throwsAssertionError);
   });
+
+  testWidgets('call dispose after removing listeners', (tester) async {
+    final notifier = _ChangeNotifierDisposeTest();
+    final widget = ChangeNotifierProvider(
+      create: (_) => notifier,
+      builder: (context, __) => Text(
+        Provider.of<_ChangeNotifierDisposeTest>(context).text,
+        textDirection: TextDirection.ltr,
+      ),
+    );
+
+    await tester.pumpWidget(widget);
+    await tester.pumpAndSettle();
+
+    // If the listener is removed after the super.dispose, this will throw
```

Comment:
That's not the case. ChangeNotfier no-longer throws if removeListener is invoked post dispose.

The assert has been removed.

Reply:
Sorry by bad, I had yet to read the custom exception bit